### DISCUSSION TITLE
DB Migration : relation "geodata_places" does not exist

### DISCUSSION THREAD

**Discussion by Rahulsharma0810** [immich-app/immich#10187] [Q&A]
### The bug

I tried to export DB to and import during server migration.

Followed: https://immich.app/docs/administration/backup-and-restore/

```
gunzip < db_dumps/last/immich-latest.sql.gz \
| sed "s/SELECT pg_catalog.set_config('search_path', '', false);/SELECT pg_catalog.set_config('search_path', 'public, pg_catalog', true);/g" \
| docker exec -i immich_postgres psql --username=postgres
```

And PostgreSQL restore was successful but I can see the below errors in the immich-server container.

Is there a way to resolve them after migration? 

### The OS that Immich Server is running on

Ubuntu 22.04

### Version of Immich Server

v1.106.1

### Version of Immich Mobile App

v1.106.1

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
N/A
```

### Your .env content

```Shell
N/A
```

### Reproduction steps

```bash
Follow the export guide and import into Postgres image "ghcr.io/tensorchord/cloudnative-pgvecto.rs:16.3"
```

### Relevant log output

```shell
[Nest] 7  - 06/11/2024, 1:49:45 PM     LOG [Microservices:SystemConfigService] LogLevel=log (set via system config)
[Nest] 7  - 06/11/2024, 1:49:45 PM     LOG [Microservices:MapRepository] Initializing metadata repository
[Nest] 7  - 06/11/2024, 1:49:45 PM   FATAL [Microservices:MapRepository] Error importing geodata
[Nest] 7  - 06/11/2024, 1:49:45 PM   FATAL [Microservices:MapRepository] QueryFailedError: relation "geodata_places" does not exist
[Nest] 7  - 06/11/2024, 1:49:45 PM   ERROR [Microservices:MetadataService] Unable to initialize reverse geocoding: QueryFailedError: relation "geodata_places" does not exist
[Nest] 7  - 06/11/2024, 1:49:45 PM     LOG [Microservices:NestFactory] Starting Nest application...
QueryFailedError: relation "geodata_places" does not exist
    at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async PostgresQueryRunner.clearTable (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:1584:9)
    at async EntityManager.clear (/usr/src/app/node_modules/typeorm/entity-manager/EntityManager.js:679:20)
    at async MapRepository.importGeodata (/usr/src/app/dist/repositories/map.repository.js:143:13)
    at async MapRepository.init (/usr/src/app/dist/repositories/map.repository.js:56:9)
    at async /usr/src/app/dist/repositories/database.repository.js:177:23
```

**Comment by jrasm91** [comment]
Do you still have a skip migrations environment variable set?

**Comment by NiklasRosenstein** [comment]
I'm currently facing the same issue, I've migrated my Immich instance to another Kubernetes cluster and Postgres database, now I get:

```
[Nest] 7  - 10/04/2024, 5:25:56 PM   ERROR [Microservices:JobService] Unable to run job handler (metadataExtraction/metadata-extraction): QueryFailedError: relation "geodata_places" does not exist
[Nest] 7  - 10/04/2024, 5:25:56 PM   ERROR [Microservices:JobService] QueryFailedError: relation "geodata_places" does not exist
    at PostgresQueryRunner.query (/usr/src/app/node_modules/typeorm/driver/postgres/PostgresQueryRunner.js:219:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async SelectQueryBuilder.loadRawResults (/usr/src/app/node_modules/typeorm/query-builder/SelectQueryBuilder.js:2192:25)
    at async SelectQueryBuilder.executeEntitiesAndRawResults (/usr/src/app/node_modules/typeorm/query-builder/SelectQueryBuilder.js:2040:26)
    at async SelectQueryBuilder.getRawAndEntities (/usr/src/app/node_modules/typeorm/query-builder/SelectQueryBuilder.js:684:29)
    at async SelectQueryBuilder.getOne (/usr/src/app/node_modules/typeorm/query-builder/SelectQueryBuilder.js:711:25)
    at async MapRepository.reverseGeocode (/usr/src/app/dist/repositories/map.repository.js:124:26)
    at async MetadataService.getGeo (/usr/src/app/dist/services/metadata.service.js:533:22)
    at async MetadataService.handleMetadataExtraction (/usr/src/app/dist/services/metadata.service.js:202:63)
    at async /usr/src/app/dist/services/job.service.js:171:36
[Nest] 7  - 10/04/2024, 5:25:56 PM   ERROR [Microservices:JobService] Object:
{
  "id": "ac9ea1ed-a172-43c5-9a09-9426d64acab2",
  "source": "upload"
}                                                                                                                                                                                                                                                                  
```

I'm not sure how to resolve this, any hint is appreciated. 🙂

**Comment by damru** [comment]
My 2 cents here, for those comfortable enough with postgres:
- run a fresh new dummy immich stack with [docker-compose](https://immich.app/docs/install/docker-compose/) (be careful to update the database image tag with the postgresql version you are actually using for your production immich instance. eg: `pg14-v0.2.0 | pg15-v0.3.0 | pg16-v0.3.0`)
- dump the `immich.geodata_places` table from the dummy instance into an sql file (with `DROP/CREATE` table requests). There are ~200k rows as-of today, so the output sql file will be ~40MB
- import it in your production `immich` database

And voila, it should be fine now.

**Reply by NiklasRosenstein** [reply]
Thanks for the suggestion! I've tried this but unfortunately I get 

```
❯ kubectl exec -n immich-personal immich-db-1 -i -- psql immich < geodata_places.sql
Defaulted container "postgres" out of: postgres, bootstrap-controller (init)
SET
SET
SET
SET
SET
     set_config
--------------------
 public, pg_catalog
(1 row)

SET
SET
SET
SET
ERROR:  missing data for column "alternateNames"
CONTEXT:  COPY geodata_places, line 8: "3040141 l'Aldosa        1.52289 42.54391        AD      04              2007-04-29      La Massana      \N"
```

The Immich instance that I sourced the `geodata_places.sql` from has the same version (v0.116.2).

```
$ kubectl exec -n default immich-db-1 -- pg_dump -a -d app -t geodata_places > geodata_places.sql
```

Note that I also had to fix the `search_path` similar [as is documented](https://immich.app/docs/administration/backup-and-restore#automatic-database-backups) to ensure it is set to

```
SELECT pg_catalog.set_config('search_path', 'public, pg_catalog', true);
```

in the dump.

**Reply by NiklasRosenstein** [reply]
This is very frustrating. 😄 The plain dump doesn't work because of the issue above; the custom format dump doesn't work because it also sets the `search_path` but I have found no way yet to change this (modifying the custom format in place doesn't work trivially as its a binary file).

**Reply by NiklasRosenstein** [reply]
Found this SO answer: https://stackoverflow.com/a/60883475 

The solution is to add `--inserts` to `pg_dump` so that it uses `INSERT INTO` instead of `COPY from stdin`.

During this quest, it appears that the `SET search_path` is fully intentional and if that causes errors like the ones described (`ERROR: type "earth" does not exist`), it appears that is an issue with the extension. 🤔 

* https://postgrespro.com/list/thread-id/2448092
* https://www.postgresql.org/message-id/03b201d0d38f%247716ee80%246544cb80%24%40com
